core/src/main/kotlin/org/evomaster/core/languagemodel/LanguageModelConnector.kt

```diff
@@ -0,0 +1,241 @@
+            )
+        }
+    }
+
+    @PreDestroy
+    private fun preDestroy() {
+        if (config.languageModelConnector) {
+            shutdown()
+        }
+    }
+
+    private fun shutdown() {
+        Lazy.assert { config.languageModelConnector }
+        workerPool.shutdown()
+        prompts.clear()
+    }
+
+    /**
+     * To query the large language server asynchronously with a simple prompt.
+     */
+    fun queryAsync(prompt: String) {
+        if (!config.languageModelConnector) {
+            return
+        }
+
+        validatePrompt(prompt)
+
+        val id = getId()
+
+        prompts[id] = Prompt(id, prompt)
+
+        val task = Runnable {
+            makeQuery(prompt)
+        }
+
+        workerPool.submit(task)
+    }
+
+    /**
+     * @return answer for the prompt as [Prompt] if exists
+     * @return null if there is no answer for the prompt
+     */
+    fun getAnswerByPrompt(prompt: String): Prompt? {
+        return prompts.filter { it.value.prompt == prompt && it.value.hasAnswer() }.values.firstOrNull()
+    }
+
+    /**
+     * @return answer for the UUID of the prompt
+     */
+    fun getAnswerById(id: String): Prompt? {
+        return prompts[id]
+    }
+
+    /**
+     * To query the large language server with a simple prompt.
+     * @return answer string from the language model server
+     */
+    fun query(prompt: String): String? {
+        if (!config.languageModelConnector) {
+            return null
+        }
+
+        validatePrompt(prompt)
+
+        return makeQuery(prompt)
+    }
+
+    private fun makeQuery(prompt: String, id: String? = null): String? {
+        validatePrompt(prompt)
+
+        val languageModelServerURL = getLanguageModelServerURL()
+
+        val languageModelName = getLanguageModelName()
+
+        val requestBody = objectMapper.writeValueAsString(
+            OllamaRequest(
+                prompt = prompt,
+                stream = false,
+                model = languageModelName.toString()
+            )
+        )
+
+        val response = call(languageModelServerURL, requestBody)
+
+        if (!id.isNullOrBlank()) {
+            if (prompts.contains(id)) {
+                val existingPrompt = prompts[id]
+
+                if (existingPrompt != null) {
+                    existingPrompt?.answer = response
+                }
+            }
+        }
+
+        return response
+    }
+
+    /**
+     * Private method to make the call to the large language model server.
+     *
+     * Note: If you are using Ollama as a server, please make sure to set the
+     * CORS origin for Ollama on the host operating system.
+     *
+     * Reference:
+     * https://medium.com/dcoderai/how-to-handle-cors-settings-in-ollama-a-comprehensive-guide-ee2a5a1beef0
+     */
+    private fun call(languageModelServerURL: String, requestBody: String): String? {
+        try {
+            val connection = URL(languageModelServerURL).openConnection() as HttpURLConnection
```

Comment:
so each time we create a new TCP socket for the calls? shouldn't be better to re-use the connection objects?

Reply:
also, might had been better to use something like `HttpClientFactory.createTrustingJerseyClient`